cpp/include/cudf/io/csv.hpp

```diff
@@ -0,0 +1,1447 @@
+   *
+   * @param val Boolean value to enable/disable.
+   */
+  void enable_windowslinetermination(bool val) { _windowslinetermination = val; }
+
+  /**
+   * @brief Sets whether to treat whitespace as field delimiter.
+   *
+   * @param val Boolean value to enable/disable.
+   */
+  void enable_delim_whitespace(bool val) { _delim_whitespace = val; }
+
+  /**
+   * @brief Sets whether to skip whitespace after the delimiter.
+   *
+   * @param val Boolean value to enable/disable.
+   */
+  void enable_skipinitialspace(bool val) { _skipinitialspace = val; }
+
+  /**
+   * @brief Sets whether to ignore empty lines or parse line values as invalid.
+   *
+   * @param val Boolean value to enable/disable.
+   */
+  void enable_skip_blank_lines(bool val) { _skip_blank_lines = val; }
+
+  /**
+   * @brief Sets quoting style.
+   *
+   * @param style Quoting style used.
+   */
+  void set_quoting(quote_style style) { _quoting = style; }
+
+  /**
+   * @brief Sets quoting character.
+   *
+   * @param ch A character to indicate quoting.
+   */
+  void set_quotechar(char ch) { _quotechar = ch; }
+
+  /**
+   * @brief Sets a quote inside a value is double-quoted.
+   *
+   * @param val Boolean value to enable/disable.
+   */
+  void enable_doublequote(bool val) { _doublequote = val; }
+
+  /**
+   * @brief Sets names of columns to read as datetime.
+   *
+   * @param col_names Vector of column names to infer as datetime.
+   */
+  void set_infer_date_names(std::vector<std::string> col_names)
+  {
+    _infer_date_names = std::move(col_names);
+  }
+
+  /**
+   * @brief Sets indexes of columns to read as datetime.
+   *
+   * @param col_names Vector of column indices to infer as datetime.
+   */
+  void set_infer_date_indexes(std::vector<int> col_ind)
+  {
+    _infer_date_indexes = std::move(col_ind);
+  }
+
+  /**
+   * @brief Sets per-column types.
+   *
+   * @param types Vector of dtypes in which the column needs to be read.
+   */
+  void set_dtypes(std::vector<std::string> types) { _dtypes = std::move(types); }
+
+  /**
+   * @brief Sets additional values to recognize as boolean true values.
+   *
+   * @param vals Vector of values to be considered to be `true`.
+   */
+  void set_true_values(std::vector<std::string> vals)
+  {
+    _true_values.insert(_true_values.end(), vals.begin(), vals.end());
+  }
+
+  /**
+   * @brief Sets additional values to recognize as boolean false values.
+   *
+   * @param vals Vector of values to be considered to be `false`.
+   */
+  void set_false_values(std::vector<std::string> vals)
+  {
+    _false_values.insert(_false_values.end(), vals.begin(), vals.end());
+  }
+
+  /**
+   * @brief Sets additional values to recognize as null values.
+   *
+   * @param vals Vector of values to be considered to be null.
+   */
+  void set_na_values(std::vector<std::string> vals)
+  {
+    if ((!vals.empty()) and (!_na_filter)) {
+      CUDF_FAIL("Can't set na_values when na_filtering is disabled");
```

Comment:
I like this approach. Just please make sure this path is tested. I think the other combinations (default_na, etc.) should already be covered.